api/src/test/java/marquez/DatasetIntegrationTest.java

```diff
@@ -357,4 +360,84 @@ public void testApp_upsertDescription() {
     // Description stays
     assertThat(dataset2.getDescription()).isEqualTo(DESCRIPTION.getDescription());
   }
+
+  @Test
+  public void testApp_doesNotShowDeletedDataset() throws IOException {
+    String namespace = "namespace";
+    String name = "table";
+    LineageEvent event =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job("namespace", "job_name", null),
+            List.of(new LineageEvent.Dataset(namespace, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    final CompletableFuture<Integer> resp =
+        this.sendLineage(Utils.toJson(event))
+            .thenApply(HttpResponse::statusCode)
+            .whenComplete(
+                (val, error) -> {
+                  if (error != null) {
+                    Assertions.fail("Could not complete request");
+                  }
+                });
+
+    // Ensure the event was correctly rejected and a proper response code returned.
+    assertThat(resp.join()).isEqualTo(201);
+
+    client.deleteDataset(namespace, name);
+
+    List<Dataset> datasets = client.listDatasets(namespace);
+    assertThat(datasets).hasSize(0);
+  }
+
+  @Test
+  public void testApp_showsDeletedDatasetAfterReceivingNewVersion() throws IOException {
+    String namespace = "namespace";
+    String name = "anotherTable";
+    LineageEvent event =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job("namespace", "job_name", null),
+            List.of(new LineageEvent.Dataset(namespace, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    CompletableFuture<Integer> resp =
+        this.sendLineage(Utils.toJson(event))
+            .thenApply(HttpResponse::statusCode)
+            .whenComplete(
+                (val, error) -> {
+                  if (error != null) {
+                    Assertions.fail("Could not complete request");
+                  }
+                });
+
+    // Ensure the event was correctly rejected and a proper response code returned.
+    assertThat(resp.join()).isEqualTo(201);
+
+    client.deleteDataset(namespace, name);
+
+    List<Dataset> datasets = client.listDatasets(namespace);
+    assertThat(datasets).hasSize(0);
+    resp =
```

Comment:
That's the tricky part. Aren't we losing here information that dataset was deleted in the past? 
If so, perhaps we should store it in `dataset_version`.

Reply:
it's not a delete of dataset but a delete action triggered from Marquez-ui. So, it does not make sense to store its history.